### PR TITLE
Kamu 115 implement block header subcomponent

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/dataset-view/dataset.service.ts [hunk 1]
```diff
@@ -52,6 +53,17 @@ export class DatasetService {
         this.datasetChanges$.next(searchDatasetInfo);
     }
 
+    private metadataBlockChanges$: Subject<MetadataBlockFragment> =
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/dataset.service.ts:56]: Let's move these extensions to a separate service, specific to `MetadataBlock` module

File: src/app/dataset-block/metadata-block/metadata-block.component.ts [hunk 2]
```diff
@@ -20,10 +27,42 @@ export class MetadataBlockComponent
...
+                    tap(() => {
+                        this.getMetadataBlock();
+                    }),
+                )
+                .subscribe(),
+            this.getMetadataBlock(),
+        );
+    }
+
+    private getMetadataBlock(): Subscription {
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/metadata-block.component.ts:63]: Maybe it's "load", not "get".

File: src/app/dataset-block/metadata-block/metadata-block.component.ts [hunk 3]
```diff
@@ -20,10 +27,42 @@ export class MetadataBlockComponent
...
+            this.activatedRoute.params
+                .pipe(pluck(ProjectLinks.URL_PARAM_BLOCK_HASH))
+                .subscribe((hash: string) => {
+                    this.blockHash = hash;
+                }),
+            this.router.events
+                .pipe(
+                    filter((event) => event instanceof NavigationEnd),
+                    tap(() => {
+                        this.getMetadataBlock();
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/metadata-block.component.ts:55]: Shouldn't this subscription be tracked too?

File: src/app/dataset-block/metadata-block/metadata-block.component.ts [hunk 4]
```diff
@@ -20,10 +27,42 @@ export class MetadataBlockComponent
...
+                    }),
+                )
+                .subscribe(),
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/metadata-block.component.ts:58]: What's the point of `tap/subscribe`, when you can simply place a handler in the subscribe callback ?

File: src/app/dataset-block/metadata-block/components/block-header/block-header.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    selector: "app-block-header",
+    templateUrl: "./block-header.component.html",
+    styleUrls: ["./block-header.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BlockHeaderComponent extends BaseComponent implements OnInit {
+    public block$: Observable<MetadataBlockFragment>;
+    @Input() public blockHash: string;
+    @Input() public datasetInfo: DatasetInfo;
+    constructor(
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/block-header/block-header.component.ts:25]: Not enough empty lines around constructor

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
And tests, tests, tests!

File: src/app/dataset-block/metadata-block/components/block-header/block-header.component.ts (hunk 5 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/block-header/block-header.component.ts:25]: Fixed

File: src/app/dataset-block/metadata-block/metadata-block.component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/metadata-block.component.ts:63]: Yes, "load" is better

File: src/app/dataset-block/metadata-block/metadata-block.component.ts (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/metadata-block.component.ts:58]: Moved function to subscribe callback

File: src/app/dataset-block/metadata-block/metadata-block.component.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/metadata-block.component.ts:55]: Yes, added this invoke to track subscription

File: src/app/dataset-view/dataset.service.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/dataset.service.ts:56]: Moved to new block.service in MetadataBlock module

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**